source/client/factories_impl.h

```diff
@@ -58,14 +61,30 @@ class OutputFormatterFactoryImpl : public OutputFormatterFactory {
 
 class RequestSourceFactoryImpl : public OptionBasedFactoryImpl, public RequestSourceFactory {
 public:
-  RequestSourceFactoryImpl(const Options& options);
+  RequestSourceFactoryImpl(const Options& options, Envoy::Api::Api& api);
   RequestSourcePtr create(const Envoy::Upstream::ClusterManagerPtr& cluster_manager,
                           Envoy::Event::Dispatcher& dispatcher, Envoy::Stats::Scope& scope,
                           absl::string_view service_cluster_name) const override;
 
+protected:
+  Envoy::Api::Api& api_;
```

Comment:
Can this be private instead or do we have a good reason to make it protected?

Reply:
I believe that this can be private.